services/server/src/server/services/storageServices/AbstractDatabaseService.ts

```diff
@@ -419,48 +419,43 @@ export default abstract class AbstractDatabaseService {
       // Start the sql transaction
       await client.query("BEGIN");
 
-      let recompiledCreationCodeInsertResult:
-        | QueryResult<Pick<DatabaseUtil.Tables.Code, "bytecode_hash">>
-        | undefined;
       let onchainCreationCodeInsertResult:
         | QueryResult<Pick<DatabaseUtil.Tables.Code, "bytecode_hash">>
         | undefined;
-      // Check if contracts_deployed needs to be updated
-      if (
-        existingVerifiedContractResult[0].transaction_hash === null &&
-        match.creatorTxHash != null &&
-        databaseColumns.onchainCreationCode
-      ) {
```

Comment:
@kuzdogan @manuelwedler I want your feedback about this. Should we insert a new contract_deployment only if the existing row doesn't have creation information or always?

I think we should **always do it** because the `verified_contracts` information are linked with the new `contract_deployment` that we are about to insert. If we skip insertion then the `verified_contract` will reference to the old `contract_deployment` that is potentially different.

Reply:
(Notice that even if we leave this "if" condition, the bug we encountered is solved)

Reply:
Resolved in a call: it's better to always update because the `contract_deployments` inserts are all idempotent.